src/commands/cicdPipeline.ts

```diff
@@ -0,0 +1,60 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import * as utils from 'util';
+import AksClusterTreeItem from '../tree/aksClusterTreeItem';
+import { IActionContext } from 'vscode-azureextensionui';
+
+const deployToAzureExtensionId = 'ms-vscode-deploy-azure.azure-deploy';
+const configurePipelineCommand = 'configure-cicd-pipeline';
+const browsePipelineCommand = 'browse-cicd-pipeline';
+
+export async function configurePipeline(context: IActionContext, target: any): Promise<void> {
+    if (await isDeployToAzureExtensionInstalled()) {
+        const cluster = await getCluster(target);
+        if (!!cluster) {
+            await executeDeployToAzureExtensionInstalled(configurePipelineCommand, cluster);
+        }
+    }
+}
+
+export async function browsePipeline(context: IActionContext, target: any): Promise<void> {
+    if (await isDeployToAzureExtensionInstalled()) {
+        const cluster = await getCluster(target);
+        if (!!cluster) {
+            await executeDeployToAzureExtensionInstalled(browsePipelineCommand, cluster);
+        }
+    }
+}
+
+async function getCluster(target: any): Promise<AksClusterTreeItem | undefined> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    if (cloudExplorer.available) {
+        const clusterTarget = cloudExplorer.api.resolveCommandTarget(target);
+        if (clusterTarget && clusterTarget.cloudName === "Azure" && clusterTarget.nodeType === "resource" && clusterTarget.cloudResource.nodeType === "cluster") {
+            return clusterTarget.cloudResource as AksClusterTreeItem;
+        } else {
+            vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+            return;
+        }
+    } else {
+        return;
+    }
+}
+
+async function isDeployToAzureExtensionInstalled(): Promise<boolean> {
+    const pipelinesExtension = vscode.extensions.getExtension(deployToAzureExtensionId);
+    if (!pipelinesExtension) {
+        vscode.window.showWarningMessage('Please install/enable `Deploy to Azure` extension and start again.');
```

Comment:
💡 regarding the flow of this info, here in line 47 we display `warning` and then we go ahead and `execute command` ? and if you see carefully we have `await` but we return `false` after execution, wondering why?

Reply:
@Tatsinnit the idea here is to open the VS Code extension marketplace page so that it is easy for user to go ahead and install the extension. That's what `extension.open` command is doing. And we are waiting for the markeplace page to open before we return from the current command.

Reply:
We are returning `false` as we can not proceed without user actually going and installing the extension. Hence we exit from current command and let user install the extension and re-trigger the command.